samples/esb/esb_monitor/src/main.c

```diff
@@ -41,27 +50,56 @@ static void leds_update(uint8_t value)
 
 	dk_set_leds(leds_mask);
 }
-#endif /* defined(CONFIG_LED_ENABLE) */
+#endif /* defined(CONFIG_LED_ENABLE) && !defined(CONFIG_ESB_SNIFFER) */
 
-static void log_packet(uint32_t timestamp)
+#if defined(CONFIG_ESB_SNIFFER)
+static void log_packet(void)
+{
```

Comment:
You can define the `log_packet` once and use `if(IS_ENABLED(CONFIG_ESB_SNIFFER))` inside it.
This avoids conditional compilation.